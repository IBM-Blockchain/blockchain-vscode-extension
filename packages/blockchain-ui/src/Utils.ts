import IVSCode from './interfaces/IVSCode';
declare const vscode: IVSCode;

class Utils {

    static changeRoute(newPath: string): void {
        dispatchEvent(new MessageEvent('message', {
            data: {
                path: newPath
            }
        }));
    }

    static postToVSCode(message: { command: string, data?: any }): void {

        if (!(window as any)['Cypress']) {
            vscode.postMessage(message);
        } else {
            window.postMessage(message, '*');
        }
    }

}

export default Utils;
