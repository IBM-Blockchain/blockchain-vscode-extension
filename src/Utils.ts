const Utils: any = {

    changeRoute(newPath: string): void {
        dispatchEvent(new MessageEvent('message', {
            data: {
                path: newPath
            }
        }));
    }

};

export default Utils;
