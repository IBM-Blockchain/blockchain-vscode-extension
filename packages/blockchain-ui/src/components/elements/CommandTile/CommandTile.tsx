import React, { Component } from 'react';
import { Tile } from 'carbon-components-react';
import HeadingCombo from '../HeadingCombo/HeadingCombo';
import Utils from '../../../Utils';
import newTabImg from '../../../resources/new-tab.svg';
import './CommandTile.scss';

interface IProps {
    title: string;
    body: string;
    command: string;
    image?: string;
}

class CommandTile extends Component<IProps> {
    constructor(props: Readonly<IProps>) {
        super(props);

        this.tileClickHandler = this.tileClickHandler.bind(this);
    }

    tileClickHandler(): void {
        Utils.postToVSCode({
            command: this.props.command
        });
    }

    render(): JSX.Element {
        return (
            <Tile id='custom-tile-hack' className='custom-tile' onClick={this.tileClickHandler}>
                <HeadingCombo
                    headingText={this.props.title}
                    subheadingText={this.props.body}
                    image={this.props.image}
                />
                <img src={newTabImg} alt='' className='new-tab-img' />
            </Tile>
        );
    }
}

export default CommandTile;
