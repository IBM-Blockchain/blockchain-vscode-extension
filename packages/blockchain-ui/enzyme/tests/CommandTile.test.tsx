import React from 'react';
import renderer from 'react-test-renderer';
import { mount, ReactWrapper } from 'enzyme';
import chai from 'chai';
import sinon from 'sinon';
import sinonChai from 'sinon-chai';

import CommandTile from '../../src/components/elements/CommandTile/CommandTile';
import Utils from '../../src/Utils';

chai.should();
chai.use(sinonChai);

interface IProps {
    title: string;
    body: string;
    options: { actionType: 'app' | 'vscode', command?: string, path?: string };
}

describe('CommandTile component', () => {
    let mySandBox: sinon.SinonSandbox;
    let postToVSCodeStub: sinon.SinonStub;
    let changeRouteStub: sinon.SinonStub;

    beforeEach(() => {
        mySandBox = sinon.createSandbox();
        postToVSCodeStub = mySandBox.stub(Utils, 'postToVSCode').resolves();
        changeRouteStub = mySandBox.stub(Utils, 'changeRoute').resolves();
    });

    afterEach(() => {
        mySandBox.restore();
    });

    it('should render the expected snapshot', () => {
        const component: any = renderer
            .create(<CommandTile title='My Tile' body='Some text I want to display in my tile' command='some_command'/>)
            .toJSON();
        expect(component).toMatchSnapshot();
    });

    it('should post to VS Code', () => {
        const component: ReactWrapper<IProps> = mount(<CommandTile title='My Tile' body='Some text I want to display in my tile' command='some_command'/>);
        component.find('.bx--tile').simulate('click');
        postToVSCodeStub.should.have.been.calledOnceWithExactly({
            command: 'some_command'
        });
    });
});
