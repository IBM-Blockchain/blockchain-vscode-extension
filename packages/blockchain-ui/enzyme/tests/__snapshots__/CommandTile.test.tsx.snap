// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CommandTile component should render the expected snapshot 1`] = `
<div
  className="bx--tile custom-tile"
  onClick={[Function]}
>
  <div
    className="heading-combo-container"
  >
    <h3>
      My Tile
    </h3>
    <p>
      Some text I want to display in my tile
    </p>
  </div>
  <img
    alt=""
    className="new-tab-img"
    src="test-file-stub"
  />
</div>
`;
